plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +103,172 @@ public void testPropertiesTable()
+            assertUpdate("DELETE FROM " + tableName + " WHERE _date =  DATE '2019-09-08'", 1);
+            assertQuerySucceeds("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 10");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 8");
+            assertQuery("SELECT count(partition._varchar) FROM \"" + tableName + "$partitions\"", "VALUES 6");
+            assertQuery("SELECT count(distinct partition._date) FROM \"" + tableName + "$partitions\"", "VALUES 2");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_varchar varchar, _date date)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', '')
+                            """);
+
+            assertQuery(
+                    "SELECT partition._varchar, partition._date FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                                ('b', DATE '2019-09-09'),
+                                ('c', DATE '2019-09-09'),
+                                ('d', DATE '2019-09-09'),
+                                ('e', DATE '2019-09-10'),
+                                ('f', DATE '2019-09-10'),
+                                (null, DATE '2019-09-10'),
+                                ('g', null),
+                                (null, null)""");
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableUnPartitioned()
+    {
+        String tableName = "test_partitions_table_unpartitioned";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE)");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE)), (1, CAST('2019-09-09' AS DATE)), (2, CAST('2019-09-09' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 6");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 0");
+            assertQueryReturnsEmptyResult("SELECT * FROM \"" + tableName + "$partitions\"");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', '')
+                            """);
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
```

Comment:
Please add test case with deletion vectors. Especially, I would like to verify whether the `size` is correct or not.

Reply:
If an `AddFileEntry` contains a DV, do we want to exclude the size in `total_size` for that `AddFileEntry`? What about for `file_count`? I think for DV, the physical size and file count does not actually change.

Reply:
Added `testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSystemTableCompatibility#testTablePartitionsWithDeletionVectors`